Able to delete post

jhoang98 opened the pull request.

jhoang98 committed:
Able to delete post

stickeedavid commented on resources/js/components/Create.vue:
```diff
@@ -88,8 +98,19 @@
+                    // this.todos.splice(id,1)
+
+                    /**VueJS**/
+                    this.$delete(this.todos, id);
```
This is using the VueJS default delete method which is different to the actions/mutators we define in the store.
You should use the store dispatch method here, calling the delete action you defined.

stickeedavid commented on resources/js/components/Create.vue:
```diff
@@ -88,8 +98,19 @@
 
+            remove(id){
+                let confirmed = confirm('Are you sure you want to delete?');
+                if (confirm){
```
You're using an undefined variable in the if statement. The variable above is named `confirmed`, not `confirm`.

Also as the value of `confirmed` won't be changing after it's been assigned, it's best to use `const` instead of `let`.

Also, just for code formatting standards, there should be a blank space after the final parenthesis in the if statement

e.g. `if (confirmed) {`

stickeedavid commented on resources/js/store/index.js:
```diff
@@ -13,12 +13,20 @@ const getters = {
+    },
+
+    DELETE: function (state, todo) {
+        state.todos.filter(todo)
```
Does ths work?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

JavaScript's array filter seems to take in a function in order to define the predicate on which to filter the collection.

Also, I don't believe JavaScript's array filter modifies in place so you'll have to reassign the value to `state.todos`

stickeedavid commented on resources/js/components/Create.vue:
```diff
@@ -1,44 +1,54 @@
+                            <td>{{todo.body}}</td>
+                            <td></td>
+                            <td><router-link class="btn btn-success" :to="{name:'Show', params: {id:todo.id}}" style="color: black">Read More</router-link>
+                                <span class="btn btn-danger" @click="remove(index)">Delete</span></td>
```
I think it'll be more robust to remove based on the todo ID rather than the index

stickeedavid commented on resources/js/components/Create.vue:
```diff
@@ -1,44 +1,54 @@
+                <div class="p-2 w-full">
+                    <div class="form-group">
+                        <label class="w-full" for="title">Title:</label>
+                        <input type="text"  class="form-control" id="title" placeholder="Enter title here" v-model="newTodoTitle" @keyup.enter="addTodoBody">
```
Extra space after `"text"` not needed

stickeedavid commented on .idea/workspace.xml:
```diff
@@ -2,21 +2,13 @@
 <project version="4">
```
This file should be added to your `.gitignore` as we don't requrie it in Git

stickeedavid commented on public/Create.js:
```diff
@@ -54,6 +54,16 @@ function _defineProperty(obj, key, value) { if (key in obj) { Object.definePrope
 //
```
This file (along with the public/js/app.js) should be in the `.gitignore` file. It's best to exclude theentire  public JS and CSS directories from the git repo as they aren't required to be submitted to git

For example, add:
`public/js`
`public/css`

stickeedavid commented on resources/js/store/index.js:
```diff
@@ -13,12 +13,20 @@ const getters = {
         commit('ADD', todo);
+    },
+
+    delete: function ({commit}, todo) {
```
If this function is only accepting the ID for the todo to remove, change the name of the parameter accordingly

jhoang98 committed:
Fixes

Used store dispatch method to call delete action.
Changed confirm to confirmed.
Changed mutation for delete function to splice.
Changed index to todo.id
Changed the name of the parameter to id

jhoang98 committed:
corrections to deleting post
